packages/shared/src/UI/components/NFTList/index.tsx

```diff
@@ -0,0 +1,231 @@
+                url={token.metadata?.imageURL ?? token.metadata?.imageURL}
+                tokenId={token.tokenId}
+                classes={{
+                    loadingFailImage: classes.loadingFailImage,
+                    iframe: classes.assetPlayerIframe,
+                    imgWrapper: classes.imgWrapper,
+                    wrapper: classes.wrapper,
+                }}
+            />
+            <Typography className={classes.caption}>#{token.tokenId}</Typography>
+        </div>
+    )
+}
+
+export const NFTList: FC<Props> = ({
+    selectable,
+    selectedPairs,
+    tokens,
+    onChange,
+    limit = 1,
+    className,
+    onNextPage,
+    finished,
+    hasError,
+}) => {
+    const { classes } = useStyles({ columns: 4 })
+
+    const isRadio = limit === 1
+    const reachedLimit = selectedPairs && selectedPairs.length >= limit
+
+    const toggleItem = useCallback(
+        (currentId: string | null, contractAddress?: string) => {
+            onChange?.(currentId, contractAddress)
+        },
+        [onChange],
+    )
+    const pluginId = useCurrentWeb3NetworkPluginID()
+    const includes: (pairs: NFTKeyPair[], pair: NFTKeyPair) => boolean =
+        pluginId === NetworkPluginID.PLUGIN_EVM
+            ? (pairs, pair) => {
+                  return !!pairs.find(([address, tokenId]) => isSameAddress(address, pair[0]) && tokenId === pair[1])
+              }
+            : (pairs, pair) => {
+                  return !!pairs.find(([, tokenId]) => tokenId === pair[1])
+              }
+
+    const SelectComponent = isRadio ? Radio : Checkbox
+    const { Others } = useWeb3State()
+
+    return (
+        <div>
+            <List className={classnames(classes.list, className)}>
+                {tokens.map((token) => {
+                    const selected = selectedPairs
+                        ? includes(selectedPairs, [token.contract?.address!, token.tokenId])
+                        : false
+                    const inactive = selectedPairs ? selectedPairs.length > 0 && !selected : false
+                    const disabled = selectable ? !isRadio && reachedLimit && !selected : false
+                    const link = token.contract
+                        ? Others?.explorerResolver?.nonFungibleTokenLink(
+                              token.contract.chainId,
+                              token.contract.address,
+                              token.tokenId,
+                          )
+                        : undefined
+                    return (
+                        <ListItem
+                            key={token.tokenId + token.id}
+                            className={classnames(classes.nftItem, {
+                                [classes.disabled]: disabled,
+                                [classes.selected]: selected,
+                                [classes.inactive]: inactive,
+                            })}>
+                            <Link
```

Comment:
There is a component called `<Linking />`. It supports rendering plain text if `link` is not available.

Reply:
Done.